tests/src/test/scala/common/rest/WskRest.scala

```diff
@@ -0,0 +1,1430 @@
+   * @param entity the name of the entity to filter from activation list
+   * @param limit the maximum number of entities to list (if entity name is not unique use Some(0))
+   * @param since (optional) only the activations since this timestamp are included
+   * @param retries the maximum retries (total timeout is retries + 1 seconds)
+   * @return activation ids found, caller must check length of sequence
+   */
+  override def pollFor(N: Int,
+                       entity: Option[String],
+                       limit: Option[Int] = Some(30),
+                       since: Option[Instant] = None,
+                       retries: Int = 10,
+                       pollPeriod: Duration = 1.second)(implicit wp: WskProps): Seq[String] = {
+    Try {
+      retry({
+        val result = idsActivation(listActivation(filter = entity, limit = limit, since = since, docs = false))
+        if (result.length >= N) result else throw PartialResult(result)
+      }, retries, waitBeforeRetry = Some(pollPeriod))
+    } match {
+      case Success(ids)                => ids
+      case Failure(PartialResult(ids)) => ids
+      case _                           => Seq()
+    }
+  }
+
+  override def get(activationId: Option[String],
+                   expectedExitCode: Int = OK.intValue,
+                   fieldFilter: Option[String] = None,
+                   last: Option[Boolean] = None)(implicit wp: WskProps): RestResult = {
+    val r = activationId match {
+      case Some(id) => {
+        val resp = requestEntity(GET, getNamePath(noun, id))
+        new RestResult(resp.status, getRespData(resp))
+      }
+      case None => {
+        new RestResult(NotFound, null)
+      }
+    }
+    validateStatusCode(expectedExitCode, r.statusCode.intValue)
+    r
+  }
+
+  /**
+   * Polls for an activation matching the given id. If found
+   * return Right(activation) else Left(result of calling REST API).
+   *
+   * @return either Left(error message) or Right(activation as JsObject)
+   */
+  override def waitForActivation(activationId: String,
+                                 initialWait: Duration = 1 second,
+                                 pollPeriod: Duration = 1 second,
+                                 totalWait: Duration = 30 seconds)(implicit wp: WskProps): Either[String, JsObject] = {
+    val activation = waitfor(() => {
+      val result = get(Some(activationId), expectedExitCode = DONTCARE_EXIT)(wp)
+      if (result.statusCode == NotFound) {
+        null
+      } else result
+    }, initialWait, pollPeriod, totalWait)
+    Try {
+      assert(activation.statusCode == OK)
+      assert(activation.getField("activationId") != "")
+      activation.respBody
+    } map {
+      Right(_)
+    } getOrElse Left(s"Cannot find activation id from '$activation'")
+
+  }
+
+  def waitForActivationConsole(totalWait: Duration = 30 seconds, sinceTime: Instant)(
+    implicit wp: WskProps): RestResult = {
+    var result = new RestResult(NotFound, null)
```

Comment:
Possible to get rid of `null` on these `new RestResult` calls?

Reply:
done.